### PR TITLE
Return list of excluded files when filtering

### PR DESCRIPTION
**Pull request by mjambon**
For https://github.com/returntocorp/semgrep/pull/3779

### CODE REVIEW

File: h_program-lang/skip_code.mli [hunk 1]
```diff
@@ -8,13 +8,28 @@ type skip =
...
+   the root).
+*)
 val filter_files:
-  skip list -> Common.dirname (* root *) -> Common.filename list ->
-  Common.filename list
+  skip list -> root:Common.dirname (* root *) -> Common.filename list ->
+  Common.filename list * Common.filename list
+
+(*
+   TODO: explain why 'root' is a list and the implementation only uses its
```
Comment (IagoAbal) [inline h_program-lang/skip_code.mli:20]: Isn't it better to explain it now?

File: h_program-lang/skip_code.ml [hunk 2]
```diff
@@ -67,8 +67,7 @@ let load file =
 (* Main entry point *)
 (*****************************************************************************)
 
-(* less: say when skipped stuff? *)
-let filter_files skip_list root xs =
+let filter_files skip_list ~root relative_paths : string list * string list =
```
Comment (IagoAbal) [inline h_program-lang/skip_code.ml:70]: Might be useful to have a dedicated (record) type to use instead of `string list * string list`.

**Review by IagoAbal — APPROVED** [review]

**Review by aryx — APPROVED** [review]

File: h_program-lang/skip_code.mli (hunk 1 above)
Comment (mjambon) [reply h_program-lang/skip_code.mli:20]: I'm not the author and I haven't figured this out. This is for @aryx.

File: h_program-lang/skip_code.ml (hunk 2 above)
Comment (mjambon) [reply h_program-lang/skip_code.ml:70]: meh. It's a yes/no split, similar to `List.partition`.

Comment (IagoAbal) [reply h_program-lang/skip_code.ml:70]: This return type is used in many places, and the type doesn't tell you what is what. IMO it is error prone and a record type is very cheap. Out of curiosity, in Core `List.partition` is called `List.partition_tf` (last time I checked at least).

Comment (IagoAbal) [reply h_program-lang/skip_code.ml:70]: It's a small detail anyways, just wanted to clarify my suggestion. For me it's always a red flag when a function takes or returns more than one built-in type `int`, `string`, `int list`, and such. I've seen quite a few bugs using such functions that would have been caught using a new type or a label.

**Pull request merged**